docker/Dockerfile

```diff
@@ -10,7 +15,9 @@ COPY . .
 RUN pip install --no-cache-dir -e .
 
 # Stage 2: Final stage
-FROM python:3.11-alpine
+FROM python:3.11-slim
+
+RUN apt-get update && apt-get install -y --no-install-recommends git
```

Comment:
This git installation is not followed by the best practice of using a multi-stage build in Docker. The purpose of multi-stage builds is to reduce image size, improve security, simplify the build process, and enhance maintainability. I have posted a proper Dockerfile in my last comment on #1159

Reply:
I don't see the comment you're linking to, @k1lgor, as you're referencing this same PR.  I'm not a Docker expert, so I'd welcome guidance if you have it.

Reply:
Here is the Docker file I have posted:
```bash
# Stage 1: Builder stage
FROM python:3.11-slim AS builder

RUN apt-get update && apt-get install -y --no-install-recommends \
  tk \
  tcl \
  curl \
  git \
  && rm -rf /var/lib/apt/lists/*

WORKDIR /app

COPY . .

RUN pip install --no-cache-dir -e .

# Stage 2: Final stage
FROM python:3.11-slim

WORKDIR /app

COPY --from=builder /usr/local/lib/python3.11/site-packages /usr/local/lib/python3.11/site-packages
COPY --from=builder /usr/local/bin /usr/local/bin
COPY --from=builder /usr/bin /usr/bin
COPY --from=builder /app .

COPY docker/entrypoint.sh .

ENTRYPOINT ["sh", "/app/entrypoint.sh"]
```
The difference is that the git is installed at the building stage instead of the final, and since git and the other packages are installed via the OS package manager, the binaries are located in the `/usr/bin/` folder and just copy that folder in the final stage. This allows `gpte` to use `git`.

Reply:
Ahhh, I see.  I've just built the container using the file you've provided and ... **it works!**

I wasn't trying to front run your solution with this PR.  I'll kindly offer a bit of reciprocal best practice advice for submitting solutions to issues.  ;)  The best practice is to submit a pull request containing your solution and then solicit feedback on the PR directly, rather than pasting code into an issues discussion.

We'd welcome this PR from you and, given that your solution predates mine, I'm happy to close this PR (and will do so)!  Will you please submit a pull request today?  We discussed this issue during our dev meeting yesterday and were keen to resolve it as it's been broken for longer than we'd like.

Reply:
One more thing -- please update the README for the Docker container in a similar manner as I have.  Thanks!

Reply:
Yeah, sure. I will prep the PR as soon as possible.